sycl/source/handler.cpp

```diff
@@ -1081,5 +1310,39 @@ handler::getCommandGraph() const {
   return MQueue->getCommandGraph();
 }
 
+sycl::detail::pi::PiImageCopyFlags
+handler::getPiImageCopyFlags(sycl::usm::alloc SrcPtrType,
+                             sycl::usm::alloc DstPtrType) {
+  if (DstPtrType == sycl::usm::alloc::device) {
+    // Dest is on device
+    if (SrcPtrType == sycl::usm::alloc::device) {
+      return sycl::detail::pi::PiImageCopyFlags::PI_IMAGE_COPY_DEVICE_TO_DEVICE;
+    } else if (SrcPtrType == sycl::usm::alloc::host ||
+               SrcPtrType == sycl::usm::alloc::unknown) {
+      return sycl::detail::pi::PiImageCopyFlags::PI_IMAGE_COPY_HOST_TO_DEVICE;
+    } else {
+      throw sycl::exception(make_error_code(errc::invalid),
+                            "Unknown copy source location");
+    }
+  } else if (DstPtrType == sycl::usm::alloc::host ||
+             DstPtrType == sycl::usm::alloc::unknown) {
+    // Dest is on host
+    if (SrcPtrType == sycl::usm::alloc::device) {
+      return sycl::detail::pi::PiImageCopyFlags::PI_IMAGE_COPY_DEVICE_TO_HOST;
+    } else if (SrcPtrType == sycl::usm::alloc::host ||
+               SrcPtrType == sycl::usm::alloc::unknown) {
+      throw sycl::exception(make_error_code(errc::invalid),
+                            "Cannot copy image from host to host");
+    } else {
+      throw sycl::exception(make_error_code(errc::invalid),
+                            "Unknown copy source location");
+    }
+  } else {
+    throw sycl::exception(make_error_code(errc::invalid),
+                          "Unknown copy destination location");
+  }
+  return {};
```

Comment:
Nit; According to the [style-guide](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return) we should not use `else` after terminating branches, so I think something like this would be better:
```suggestion
  if (DstPtrType == sycl::usm::alloc::device) {
    // Dest is on device
    if (SrcPtrType == sycl::usm::alloc::device)
      return sycl::detail::pi::PiImageCopyFlags::PI_IMAGE_COPY_DEVICE_TO_DEVICE;
    if (SrcPtrType == sycl::usm::alloc::host ||
        SrcPtrType == sycl::usm::alloc::unknown)
      return sycl::detail::pi::PiImageCopyFlags::PI_IMAGE_COPY_HOST_TO_DEVICE;
    throw sycl::exception(make_error_code(errc::invalid),
                          "Unknown copy source location");
  }
  if (DstPtrType == sycl::usm::alloc::host ||
      DstPtrType == sycl::usm::alloc::unknown) {
    // Dest is on host
    if (SrcPtrType == sycl::usm::alloc::device)
      return sycl::detail::pi::PiImageCopyFlags::PI_IMAGE_COPY_DEVICE_TO_HOST;
    if (SrcPtrType == sycl::usm::alloc::host ||
        SrcPtrType == sycl::usm::alloc::unknown)
      throw sycl::exception(make_error_code(errc::invalid),
                            "Cannot copy image from host to host");
    throw sycl::exception(make_error_code(errc::invalid),
                          "Unknown copy source location");
  }
  throw sycl::exception(make_error_code(errc::invalid),
                        "Unknown copy destination location");
```

Reply:
Fixed.